cmd/configure_test.go

```diff
@@ -7,76 +7,86 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type testDefinition struct {
+	desc           string
+	args           []string
+	existingUsrCfg *config.UserConfig
+	expectedUsrCfg *config.UserConfig
+	existingAPICfg *config.APIConfig
+	expectedAPICfg *config.APIConfig
+}
+
 func TestConfigure(t *testing.T) {
-	tests := []struct {
-		desc           string
-		args           []string
-		existingUsrCfg *config.UserConfig
-		expectedUsrCfg *config.UserConfig
-		existingAPICfg *config.APIConfig
-		expectedAPICfg *config.APIConfig
-	}{
-		{
+	tests := []testDefinition{
+		testDefinition{
 			desc:           "It writes the flags when there is no config file.",
 			args:           []string{"fakeapp", "configure", "--token", "a", "--workspace", "/a", "--api", "http://example.com"},
 			existingUsrCfg: nil,
 			expectedUsrCfg: &config.UserConfig{Token: "a", Workspace: "/a"},
 			existingAPICfg: nil,
 			expectedAPICfg: &config.APIConfig{BaseURL: "http://example.com"},
 		},
-		{
+		testDefinition{
 			desc:           "It overwrites the flags in the config file.",
 			args:           []string{"fakeapp", "configure", "--token", "b", "--workspace", "/b", "--api", "http://example.com/v2"},
 			existingUsrCfg: &config.UserConfig{Token: "token-b", Workspace: "/workspace-b"},
 			expectedUsrCfg: &config.UserConfig{Token: "b", Workspace: "/b"},
 			existingAPICfg: &config.APIConfig{BaseURL: "http://example.com/v1"},
 			expectedAPICfg: &config.APIConfig{BaseURL: "http://example.com/v2"},
 		},
-		{
+		testDefinition{
 			desc:           "It overwrites the flags that are passed, without losing the ones that are not.",
 			args:           []string{"fakeapp", "configure", "--token", "c"},
 			existingUsrCfg: &config.UserConfig{Token: "token-c", Workspace: "/workspace-c"},
 			expectedUsrCfg: &config.UserConfig{Token: "c", Workspace: "/workspace-c"},
 		},
-		{
+		testDefinition{
 			desc:           "It gets the default API base URL.",
 			args:           []string{"fakeapp", "configure"},
 			existingAPICfg: &config.APIConfig{},
 			expectedAPICfg: &config.APIConfig{BaseURL: "https://v2.exercism.io/api/v1"},
 		},
 	}
 
-	for _, test := range tests {
-		cmdTest := &CommandTest{
+	for _, definition := range tests {
```

Comment:
It’s hard to make the distinction on the context of definition as we enter the makeTest func. Maybe it’s because I’m used to seeing it but testcase seems easier to wrap my head around.

Reply:
I agree with all you comment and made the change locally. Will commit soon, thank you.

Also in the meantime I needed to do the same modification on another test (TestDetectPathType). Would you rather avec separate branch/PR for each test that can benefit from this pattern, or separate commit all in the same branch ? (there are 21 tests using testCases)

Reply:
I wouldn't say favor. We generally lean towards readability and consistency. `xx(testCase testCase)` stomps out the type so I would not recommend that. Seeing as makeFunc is essentially a method of testCase, which could also be written as ```func (tc testCase) makeTest() func(*t.Testing) ``` I would think that `tc` as the name is sufficient, an alternative is `testcase` as a mixed case is not an option here.

Reply:
> Also in the meantime I needed to do the same modification on another test (TestDetectPathType). Would you rather avec separate branch/PR for each test that can benefit from this pattern, or separate commit all in the same branch ? (there are 21 tests using testCases)

If they are the same type of change than one PR can work. I ask that you commits be broken down per test file (no mixing between configure_test, workspace_test, etc) so that we can review each commit without having to revisit already reviewed items. Please let me know what you think or my ask is not clear. 

Thanks again for all your effort with the CLI.